velox/exec/tests/TaskTest.cpp

```diff
@@ -627,6 +628,42 @@ TEST_F(TaskTest, supportsSingleThreadedExecution) {
   ASSERT_FALSE(task->supportsSingleThreadedExecution());
 }
 
+TEST_F(TaskTest, updateBroadCastOutputBuffers) {
+  auto plan = PlanBuilder()
+                  .tableScan(ROW({"c0"}, {BIGINT()}))
+                  .project({"c0 % 10"})
+                  .partitionedOutputBroadcast(std::vector<std::string>{"p0"})
+                  .planFragment();
+  auto taskId = "single.execution.task.0";
+  auto task = std::make_shared<exec::Task>(
+      taskId, plan, 0, std::make_shared<core::QueryCtx>(driverExecutor_.get()));
+  auto bufferManager = PartitionedOutputBufferManager::getInstance();
+  {
+    auto bufferMgrShared = bufferManager.lock();
+    bufferMgrShared->initializeTask(task, true, 2, 2);
+  }
+  ASSERT_EQ(
+      task->updateBroadcastOutputBuffers(10, false),
+      UpdateBroadcastStatus::SUCCESS);
+
+  {
+    // Mock task cancellation.
+    task->requestCancel();
+    auto bufferMgrShared = bufferManager.lock();
+    bufferMgrShared->removeTask(taskId);
+  }
+
+  // Try update after task was cancelled and buffer for that task was removed.
+  ASSERT_EQ(
+      task->updateBroadcastOutputBuffers(11, true),
+      UpdateBroadcastStatus::BUFFERS_NOT_FOUND);
+
+  // Returns NO_OP as the previous call set noMoreBuffers to true.
```

Comment:
Maybe we should separate the test for BUFFERS_NOT_FOUND from that for NO_OP. Is that possible ?

Reply:
I have re-ordered the tests and isolated in code-blocks.